source/Halibut/Transport/Protocol/ControlMessageReader.cs

```diff
@@ -151,5 +159,15 @@ static CancellationTokenSource GetCancellationTokenSourceFromStreamReadTimeout(S
 
             return new CancellationTokenSource(HalibutLimits.TcpClientReceiveTimeout); // Just default to a higher timeout, rather than be cancellation token none
         }
+        
+        CancellationTokenSource GetCancellationTokenSourceFromStreamReadTimeoutAsync(Stream stream)
+        {
+            if (stream.CanTimeout)
```

Comment:
TODO mark this as TODO to note we should always be given a stream that can timeout.

Reply:
WebSockets?

Reply:
We were not thinking of wrapping that stream in a stream that fakes timing out?